ValveKeyValue/ValveKeyValue/StringTable.cs

```diff
@@ -0,0 +1,58 @@
+﻿namespace ValveKeyValue
+{
+    public sealed class StringTable
+    {
+        public StringTable(Memory<string> values)
+        {
+            this.lookup = values;
+        }
+
+        readonly Memory<string> lookup;
+        Dictionary<string, int> reverse;
+
+        public string this[int index]
+        {
+            get
+            {
+                if (index < 0)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(index), "Index must be non-negative.");
+                }
+
+                if (index >= lookup.Length)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(index), index, "Index must be less than the number of strings in the table.");
+                }
+
+                return lookup.Span[index];
+            }
+        }
+
+        public int IndexOf(string value)
```

Comment:
Indexof usually implies O(n), should this be this[string]?

Reply:
Never heard of that before, but yes.